### PR TITLE
Fix non-quantized QConv2DBatchnorm conversion

### LINKED ISSUE #928: Non-quantized QKeras layers break conversion
## Quick summary
While a bit nonsensical, QKeras layers without quantizers are legal. For example `QDense(n_units)` (without `kernel_quantizer=something`) fails conversion since we expect the quantizers to be present in the converter. In fact, the IR instance of `Dense` doesn't need a quantizer, so it is just the converter that is problematic.

### Steps to Reproduce

Create any model with QKeras layers and don't specify quantizers.

### Expected behavior
The model should convert to IR.

### Actual behavior
The converter fails.

### Possible fix
We need to go through `hls4ml/converters/keras/qkeras.py` and check if supported layers pass without the quantizers specified. The check already exists for bias, we should add it for weights.

### Additional context
I wish this was caught on QKeras side and creation of quantized models without a quantizer is prohibited, but that's out of our control.

### PR DESCRIPTION
**Pull request by hadronic-arvind**
## Summary

- Allow `QConv2DBatchnorm` layers without weight or bias quantizers to convert after batch-normalization folding.
- Preserve folded unquantized weights and bias when their corresponding quantizer is `None`.
- Add matched Keras 2/QKeras and Keras 3/QKeras-v3 regressions for both `Latency` and `Resource` strategies.

Fixes #928.

## Tests

- `pre-commit run --files hls4ml/model/layers.py test/pytest/test_qkeras.py test/pytest/test_qkerasV3.py`
- Keras 2/QKeras focused regression: 2 passed.
- Keras 3/QKeras-v3 focused regression: 2 passed.
- Keras 2/QKeras cross-backend acceptance subset: 14 passed (Vivado, Vitis, Quartus; `io_parallel` and `io_stream`).
- Keras 3/QKeras-v3 cross-backend acceptance subset: 14 passed (Vivado, Vitis, Quartus; `io_parallel` and `io_stream`).
- Conversion-only validation passed for Vivado, Vitis, Quartus, Catapult, and oneAPI, including the Vivado `Resource` path.

### CODE REVIEW

**Comment by vloncar** [comment]
Do you actually have a model with QConv2DBatchnorm that requires this? We don't actively support this layer anymore and it will likely be removed in the future.

Rule of thumb: inquire about an issue before blindly telling your agent to solve it.
